src/main/java/xyz/kyngs/herbot/handlers/user/UserHandler.java

```diff
@@ -0,0 +1,64 @@
+package xyz.kyngs.herbot.handlers.user;
+
+import xyz.kyngs.herbot.HerBot;
+import xyz.kyngs.herbot.handlers.AbstractHandler;
+import xyz.kyngs.herbot.util.ListUtil;
+
+import java.sql.Date;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+public class UserHandler extends AbstractHandler {
+
+    private final Map<String, UserProfile> cache;
+
+    public UserHandler(HerBot herBot) {
+        super(herBot);
+        cache = new HashMap<>();
+    }
+
+    public UserProfile getUser(String userID) {
+        return cache.computeIfAbsent(userID, k -> {
+            var databaseManager = herBot.getDatabaseManager();
+            var future = new CompletableFuture<UserProfile>();
+            databaseManager.getMySQL().sync().schedule(connection -> {
+                var ps = connection.prepareStatement("SELECT coins, xp, daily_streak, last_daily_claim, permissions FROM user_data WHERE user_id=?");
+                ps.setString(1, userID);
+                var rs = ps.executeQuery();
+
+                if (!rs.next()) {
+                    var user = new UserProfile(5000, 0, -1, LocalDate.now().minusDays(1), new ArrayList<>());
+                    saveUser(user, userID);
+                    future.complete(user);
+                    return;
+                }
+
+                var user = new UserProfile(rs.getLong("coins"), rs.getLong("xp"), rs.getInt("daily_streak"), rs.getDate("last_daily_claim").toLocalDate(), ListUtil.deserializeList(rs.getString("permissions")));
+
+                future.complete(user);
+
+            });
+            return future.getNow(null);
+        });
+    }
+
+    public void saveUser(UserProfile userProfile, String userID) {
+        herBot.getDatabaseManager().getMySQL().sync().schedule(connection -> {
+            var ps = connection.prepareStatement("REPLACE user_data(coins, user_id, xp, daily_streak, last_daily_claim, permissions) VALUES (?, ?, ?, ?, ?, ?) ");
```

Comment:
Isn't update the function you should use? https://www.w3schools.com/sql/sql_update.asp

Reply:
Nope cause update does not work if the user is not in database, replace basically inserts it if it is missing and updates if the user is already in db

Reply:
I still feel that this my cause some trobles, but ok.